python-package/lightgbm/basic.py

```diff
@@ -2330,7 +2331,7 @@ def _reverse_update_params(self) -> "Dataset":
     def set_field(
         self,
         field_name: str,
-        data
+        data: Optional[Union[List[float], List[int], np.ndarray, pd_Series, pd_DataFrame]]
```

Comment:
This can also be list of lists

Reply:
```suggestion
        data: Optional[Union[List[List[float]], List[List[int]], List[float], List[int], np.ndarray, pd_Series, pd_DataFrame]]
```

Ohhhhhhhh for multi-class `init_score`, right! Missed that when I read through the docstring, thanks for catching it.